src/refdb_fs.c

```diff
@@ -362,6 +364,13 @@ static const char *loose_parse_symbolic(git_buf *file_content)
 	return refname_start;
 }
 
+static bool is_per_worktree_ref(const char *ref_name)
+{
+	return strcmp("HEAD", ref_name) == 0 ||
+	    strcmp("MERGE_HEAD", ref_name) == 0 ||
+	    strcmp("ORIG_HEAD", ref_name) == 0;
```

Comment:
For now I'd rather stick with how upstream git.git treats this for simplicity. As soon as this is merged we may still extend the API to also allow custom refs to be per worktree. Added the missing FETCH_HEAD, though.